starter_code/express-spotify-app/app.js

```diff
@@ -0,0 +1,87 @@
+const SpotifyWebApi = require("spotify-web-api-node");
+const express = require("express");
+const app = express();
+const hbs = require("hbs");
+const path = require("path");
+const bodyParser = require("body-parser");
+
+app.set("views", path.join(__dirname, "views"));
+app.set("view engine", "hbs");
+
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(express.static(path.join(__dirname, 'public')));
+hbs.registerPartials(path.join(__dirname, '/views/partials'));
+
+// Remember to paste your credentials here
+const clientId = "fcbe5c9bafc44950b2f50f4ca18fb7e2",
+  clientSecret = "8ca60a7ae9764e769746e05f7fb4850c";
+
+const spotifyApi = new SpotifyWebApi({
+  clientId: clientId,
+  clientSecret: clientSecret
+});
+
+app.get("/", function(req, res) {
+  res.render("index");
+});
+
+app.get("/artists", function(req, res) {
+	const {artist} = req.query;
+
+	const queryArtist = spotifyApi.searchArtists(artist);
+	queryArtist.then(data => {
+			let artistList = data.body.artists.items;
+			console.log(data)
+			res.render('artists', {artistList});
+    })
+    .catch(err => {
+      // ----> 'HERE WE CAPTURE THE ERROR'
```

Comment:
console.log(err) or res.render('error', err); => in case you have an error.hbs template

Reply:
oh!